include/ThemeManager.hpp

```diff
@@ -0,0 +1,35 @@
+/*
+** Project FileShare, 2022
+**
+** Author Léo Lhuile
+**
+** Started on  Thu Jul 24 15:44:13 2025 Léo Lhuile
+** Last update Thu Jul 24 15:44:13 2025 Léo Lhuile
+**
+** ThemeManager.hpp : Theme management system
+*/
+
+#pragma once
+
+#include <string>
+#include <TGUI/TGUI.hpp>
+
+namespace FileShare::GUI {
+    class ThemeManager {
+        public:
+            static ThemeManager &getInstance();
+
+            void setTheme(const std::string &theme);
+            std::string getCurrentTheme() const { return this->currentTheme; }
+
+            std::string getIconThemeSuffix() const { return this->currentTheme == "light" ? "_black" : "_white"; }
```

Comment:
Propositon, instead of having to do `"file_path" + suffix + ".extension" everywhere we use this`, could we have something like : 
```suggestion
            std::string getThemedAssetPath(std::string filename, std::string extension = ".svg") const { return "assets/images" + filename + (this->currentTheme == "light" ? "_black" : "_white") + extension; }
```

Reply:
also, I believe we should move the images in their own folder (eg: `assets/images/light/` and `assets/images/dark/`) and we can rename them to remove the `_white` / `_black`.

Making sure we use the name of the theme for the folder, makes it so that we dont need that light -> white / black -> dark` ternary.